### PR TITLE
Adding support for #public

### PR DESCRIPTION
**Pull request by mwoc**
Solves #25 

I think I got it right this time, but if you can construct a failing test please let me know.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -21,7 +22,59 @@ module.exports.load = function load(rootFileName, options) {
 
     var loadedAbsoluteFileNames = []; // Do some housekeeping to avoid cycles
 
-    return (function readFileAndResolveIncludes(fileName) {
+    // Have to always keep public keys separate within resolvePublic
+    // If a private key is overwritten by a public key, it's an error, and vice-versa. Name your config keys something else!
+    // Empty object leaves and non-public keys are filtered out when requesting with public:true
+
+    function resolvePublic(obj, isToplevel, publicOnly) {
```
Comment (papandreou) [inline lib/index.js]: Why accept `publicOnly` as a parameter here? You can nuke it (along with the `var publicOnly` above, and just use `options.publicOnly`.

Comment (papandreou) [inline lib/index.js]: Ah, withdrawn, you pass `true` and `false` below.

I'd still get rid of the `var publicOnly = ...` above, though.

File: lib/index.js [hunk 2]
```diff
@@ -21,7 +22,59 @@ module.exports.load = function load(rootFileName, options) {
...
+                        publicObj[key] = resolvedPublic;
+                    }
+
+                    obj[key] = resolvePublic(obj[key], false, false);
+                }
+            });
+
+            if (typeof obj['#public'] !== 'undefined') {
+                if (!(typeof obj['#public'] === 'object' && obj['#public'] !== null)) {
+                    throw new Error('#public must always be an object');
```
Comment (papandreou) [inline lib/index.js]: arrays are also supported, right?

**Comment by papandreou** [comment]
:+1:, except for those minor nits. Also, please squash the commits a bit :)

File: lib/index.js (hunk 2 above)
Comment (mwoc) [inline lib/index.js]: Will make a test for that and update error msg

Comment (mwoc) [inline lib/index.js]: Arrays are not supported anymore, as the behaviour was not useful. Have added a check and a test for that now.

File: lib/index.js (hunk 1 above)
Comment (mwoc) [inline lib/index.js]: Done https://github.com/One-com/node-oconf/commit/13eb63a9b26f0001913810fea26d5244597e9201

**Comment by mwoc** [comment]
Got rid of 11 commits by squashing, does it look mergable now?

**Comment by papandreou** [comment]
:+1:

**Pull request merged**